examples/hello-grudge.py

```diff
@@ -0,0 +1,117 @@
+import numpy as np
+import pyopencl as cl
+from grudge.discretization import DiscretizationCollection
+import grudge.op as op
+from meshmode.mesh.generation import generate_box_mesh
+from meshmode.array_context import PyOpenCLArrayContext
+from arraycontext import freeze, thaw
+from grudge.dof_desc import DTAG_BOUNDARY, FACE_RESTR_INTERIOR
+
+
+ctx = cl.create_some_context()
+queue = cl.CommandQueue(ctx)
+actx = PyOpenCLArrayContext(queue)
+
+nel = 10
+coords = np.linspace(0, 2*np.pi, nel)
+mesh = generate_box_mesh((coords,),
+                         boundary_tag_to_face={"left": ["-x"],
+                                               "right": ["+x"]})
+dcoll = DiscretizationCollection(actx, mesh, order=1)
+
+
+def initial_condition(x):
+    # 'x' contains ndim arrays.
+    # 'x[0]' gets the first coordinate value of all the nodes
+    return actx.np.sin(x[0])
+
+
+def left_boundary_condition(x, t):
+    return actx.np.sin(x[0] - 2 * np.pi * t)
+
+
+def flux(dcoll, u_tpair):
+    dd = u_tpair.dd
+    velocity = np.array([2 * np.pi])
+    normal = thaw(dcoll.normal(dd), actx)
+
+    v_dot_n = np.dot(velocity, normal)
+    u_upwind = actx.np.where(v_dot_n > 0,
+                             u_tpair.int, u_tpair.ext)
+    return u_upwind * v_dot_n
+
+
+vol_discr = dcoll.discr_from_dd("vol")
+left_bndry = DTAG_BOUNDARY("left")
+right_bndry = DTAG_BOUNDARY("right")
+
+x_vol = thaw(dcoll.nodes(), actx)
+x_bndry = thaw(dcoll.discr_from_dd(left_bndry).nodes(), actx)
+
+uh = initial_condition(x_vol)
+
+dt = 0.001
+t = 0
+t_final = 0.5
+
+# timestepper loop
+while t < t_final:
+    uh = thaw(freeze(uh, actx), actx)
```

Comment:
Why is this here? (I'm guessing something to do with lazy? If so, comment.

Reply:
Since this was really only needed for the lazy mode, avoided it to keep things simple. (Thanks!)